### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by wacoo**
- Fix code to prevent N+1 query problem
- Add integration test for user
- Add integration test for post

Screenshots
==========
Before
=====
bf

After
====
af

### CODE REVIEW

File: spec/user_integraion_spec.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      expect(page).to have_current_path(user_path(user1))
+      visit user_path(@user2)
+      click_link 'Lily'
+      expect(page).to have_current_path(user_path(@user2))
+    end
+  end
+  context 'show' do
+    scenario 'profile should display the correct images' do
+      visit users_path(user1)
+      expect(page).to have_css('img[src="https://media.giphy.com/media/tZaFa1m8UfzXy/giphy.gif"]')
```
Comment (ghost) [inline spec/user_integraion_spec.rb]: - [ ] on your `user#show` tests, Kindly add :
  - Tests that asserts you can see the user's first 3 posts.

File: spec/post_integraion_spec.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+end
+
+RSpec.describe 'Post', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let(:user1) do
+    User.create(name: 'John', photo: 'https://media.giphy.com/media/tZaFa1m8UfzXy/giphy.gif',
+                bio: 'This is John\'s bio', posts_counter: 0)
```
Comment (ghost) [inline spec/post_integraion_spec.rb]: - [ ] on your `post#index` tests, Kindly add :
  - Tests that asserts you can see a section for pagination if there are more posts than fit on the view.

File: spec/post_integraion_spec.rb [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                         likes_counter: 0)
+    Post.create(author: user1, title: 'Name of the earth', text: 'The book of earth.', comments_counter: 0,
+                likes_counter: 0)
+    Post.create(author: user1, title: 'Name of the water',
+                text: 'The book of water and book of water book of water book of water book of water book of water' \
+                      'book of water book of water book of water book of water book of water book of water book of' \
+                      'water book of water book of water book of water', comments_counter: 0, likes_counter: 0)
+
+    @cmt1 = Comment.create(user: @user2, post: @post2, text: 'Hey nice post!')
+    @cmt2 = Comment.create(user: @user2, post: @post2, text: 'Hey great post!')
```
Comment (ghost) [inline spec/post_integraion_spec.rb]: #### 🌟 Optional Suggestion 

- [ ] Consider organizing your feature tests within a feature folder. For each feature (e.g., Users), create a dedicated folder (e.g., users) and inside it, have `index`and `show` files to test corresponding views. This approach enhances code organization, separates concerns, and makes maintenance a breeze! 🚀

#### Example
```
📁 spec
  📁 features
    📁 users
      📄 index_spec.rb
      📄 show_spec.rb
    📁 posts
      📄 index_spec.rb
      📄 show_spec.rb


```

#### Keep up with the good work! 🌟✨

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Tested Most scenarios
✅Polished and Professional documentation
✅ Fixed the `N+1` problem👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on addressing the requested changes from the previous reviewer

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**